float32 orientation
float32 xPos
float32 zPos